### ISSUE TITLE
hotspot only groups

### ISSUE THREAD

**Issue by sp5lg** [n7tae/smart-group-server#12]
Hello,
This is probably a feature request. Since misusing public repeaters and joining to routing group "might make you unpopular", would you consider making this a configuration option? One could mark a group in config as "hotspot only" so that sgs will only do zone routing to a call if it's repeater is a match. Wouldn't that solve the problem of using public repeaters for routing? SGS knows the route and is the only instance in the middle that can stop it.

**Comment by n7tae** [comment]
But what are you matching to when you say "will only do routing if it's a match"? QuadNet currently has about 5300 gateways logged in at any time (this number grows each month) and over 15,000 repeater modules have been used on QuadNet in the past 90 days. There is no easy way to tell the difference between a hot-spot and a public repeater.

**Comment by sp5lg** [comment]
Right, so smart group has callsigns signed up for it. SGS needs to know callsign and it’s repeater to send stream. Callsign is using hotspot if 7 position substring of callsign and repeater is equal. Of course it’s possible to have it different but this is the simplest and the most common situation. 7 chars of MYCALL, RPT1 and RPT2 are equal - it’s a hotspot.

**Comment by w1bsb** [comment]
Some users have multiple hotspots that use club call signs which are vastly different than what their MYCALL would be, so this would not cover that scenario.  Or a user using another person’s hotspot with their permission.  It also doesn’t cover countries that will issue multiple calls to people, like Canada, who choose to use a unique call sign for their hotspot instead of their own call sign.

It would require something a bit more complex, I’m afraid.

Colby

**Comment by sp5lg** [comment]
Well, maybe those are exceptions? I myself am using such a secondary call sign for hotspot but I’m willing to give up for the comfort of smart routing. I’m voting for parameter - let the group and server administrator decide. I was thinking about why Smart Groups have such a small adoption and one of those reasons is probably that administrators are afraid of people misusing smart routing. And this is actually true. Right now administrator has no control over the streams his own server will send. Please, consider a choice of parameter over no control and freedom of potential users.

**Comment by w1bsb** [comment]
I'm a bit confused, because if we add the option to the smart group server itself, it doesn't help the repeater owners at all since the group administrator can determine at his will if he wants to allow his group to route to repeaters or not.  Repeater owners already have control over routing. They can choose not to do it by not opening UDP Port 40000 on their repeater. Additionally, repeater owners could theoretically block routing from any smart group they wanted by blocking its IP Address, which is publicly available via the IRC system.

I think wide spread adoption of routing is slow simply because there has been a huge push for reflectors. Take the OpenSpot for example, which offers zero routing capability, and only reflector linking. 

I don't think the feature your requesting would have the ultimate desired effect.  If I'm not mistaken, you want a "hotspot" option that disallows rouing to the group unless MYCALL, RPT1, and RPT2 match?

**Comment by sp5lg** [comment]
Yes, this request is exactly about making it possible to configure SGS to transmit only to so called SelfOnly hotspots (MYCALL=RPT). This will allow SGS administrator to shape the streams it pushes to the network.

I understand your point. But taking it one by one:
1. If SGS administrator decides to not filter and push unwanted stream to public repeater, that repeater admin can blacklist SGS server. In case repeater is getting unwanted streams, that’s the only quick option to stop it. Or wait for timeout (a few hours maybe?).
2. I’m assuming repeater owners are kind and nice people. They don’t want to stop routing just because. And I’m sure, you/we don’t want them to do that either. But as SGS administrator I might force them to block my SGS since right now there is no other way to shape the stream flow from my server. In case it is misused.

My perspective it this: I want to listen to smart groups on my Dstar radio because it allows me to listen to multiple reflectors, especially multi mode ones. To do that, I need SGS server. If I put up public SGS server, I have no control over the streams my server is pushing toward public Dstar repeaters. And I’m afraid of it! Mad repeater admins block my SGS server and that’s it, end of routing story. And I want to be kind and nice to those repeater administrators too!

I think, Smart Groups is one hell of a Dstar feature! Combined with multi mode reflectors, networks are following you and your radio! So why there is only 10 opened user sessions right now worldwide? Maybe Dstar users are also kind and nice and are a bit afraid of routing? Small number of groups - this is true. Maybe potential SGS administrators are afraid that anybody can misuse their server? Otherwise, why not have SGS side by side with every reflector? So I could choose to connect or ask smart group to push stream from that reflector. Why not?

https://www.openquad.net/faq.php
“I heard that routing was bad. Why does QuadNet allow it?”
“Is there anyway me doing routing can cause someone else problems?”
SelfOnly parameter for smart group would make all these problems go away. Don’t you think? Yes it would also disable “follow me” on public repeaters. Small price to pay since using hotspot is still the recommended solution.

**Comment by n7tae** [comment]
I appreciate your enthusiasm but your solution comes a too high a price. I know several users that configure their hot-spots with different callsigns, and I don't get out that much! I believe it's unrealistic to expect an SGS admin to maintain a list of exceptions and it will be very confusing for some new sgs users if they don't. (I can route on DSTAR1, but how come I can't route on ALABAMA?)
I wrote the faq to try to give an honest assessment of why some hams are hell-bent against routing, but the scenarios I described don't actually happen all that often, and in fact, they are quite rare.
As Colby said, public repeater owners can easily prevent routing on their system by not opening routing ports, but I would go further: If they don't want to allow routing, why are they even using IRC? They shouldn't even be logging into QuadNet if they don't intend on supporting routing.
Finally, I think this would have to be implemented in the routine that processes incoming D-Star header packets and this class method is already chunky enough without having to also have to compare MyCall, Rptr1 and Rpt2 and possibly look up an exception rule.

**Comment by sp5lg** [comment]
I was thinking about it, and SGS already has all the tools.
1. Exception list: make it a dedicated smart group, so anybody can sign up to be routed anywhere
2. Notify users: same as with LOGON and LOGOFF, send message “PUBLIC RPTR? SIGN UP TO BE ROUTED”
And above all, this parameter would be for the SGS administrator to decide, not mandatory.
Well, at least I tried :) Thanks anyway!
